### ISSUE TITLE
Filename search on Object Storage Bucket list page

### ISSUE THREAD

**Issue by Oloryn** [linode/manager#6257]
One thing I'm missing on the the page that lists the objects in a bucket is some means to doing a search on the file names.  With the current implementation, if you have a lot of objects in a bucket, you have to force the full list to full length by repeatedly going to the bottom of the page, then do a find-in-page in the browser.  Having a way of restricting the files shown to a pattern would be handy.

**Comment by johnwcallahan** [comment]
Thanks for opening this issue, @Oloryn. 

We've talked about adding a "filter" text input near the top of the object listing page, which would allow filtering for a specific object.  A possible limitation is that results would be limited to exact matches. 

So, from the "root" of the bucket, typing in `hello` would match:

`hello.txt`
`hello_world.txt`

But it would _not_ match:

`a_hello.txt`
`folder1/hello.txt`

Would something like that align with your use case?

**Comment by Oloryn** [comment]
So, it's essentially a 'must start with' filter?  For my uses, that would work nicely.

**Comment by johnwcallahan** [comment]
Exactly. We're still in the requirements gathering phase for this feature, so it's helpful to know this implementation would work well for your use case. 

I'll leave this issue open for the time being and update it when there are further developments.

**Comment by Oloryn** [comment]
Sorry about that.  I was fiddling with the Github app on my phone and mistakenly hit the close button on this issue.

**Comment by Oloryn** [comment]
Any plans on doing this soon?  This had been idle for a couple of years.

**Comment by johnwcallahan** [comment]
Hey @Oloryn, thanks for your patience. This is still in our backlog, but other features are currently higher in priority.

As a workaround, [s3cmd](https://www.linode.com/docs/products/storage/object-storage/guides/s3cmd/) offers a way to filter by prefix:

```
$ s3cmd ls s3://my-bucket/path/to/file/hello

2022-02-18 04:01       614256  s3://my-bucket/path/to/file/hello_world.txt
2022-02-18 04:01       184076  s3://my-bucket/path/to/file/helloworld.txt
2022-02-18 04:01       312952  s3://my-bucket/path/to/file/helloworld
```
